### DISCUSSION TITLE
ebitenmobile bind fails in a temporary project on go mod tidy -v

### DISCUSSION THREAD

**Discussion by kordax** [hajimehoshi/ebiten#2045] [Q&A]
`ebitenmobile bind -x -v -target android -javapkg net.isoftservices.client -o ./android/app/libs/client.aar octopus/parts/webgames/bin/shooters-go-client/src/android/
`
results in:

> PATH=/tmp/ebitenmobile-18041009/bin:/home/kordax/.nvm/versions/node/v17.6.0/bin:/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/usr/local/go/bin:/home/kordax/go/bin:/home/kordax/Android/Sdk/platform-tools/
> cd /tmp/ebitenmobile-18041009
> GO111MODULE=on go mod init ebitenmobiletemporary
> GO111MODULE=on go get golang.org/x/mobile/cmd/gomobile@5d9a33257ab559d10fa8c96087aed99bf8b6d868
> GO111MODULE=on go mod tidy
> GO111MODULE=on go build -o bin/gomobile golang.org/x/mobile/cmd/gomobile
> GO111MODULE=on go build -o bin/gobind-original golang.org/x/mobile/cmd/gobind
> GO111MODULE=on go build -o bin/gobind -tags ebitenmobilegobind src/gobind.go
> gomobile init
> GOMOBILE=/home/kordax/go/pkg/gomobile
> WORK=/tmp/gomobile-work-1926535437
> GOOS=android CGO_ENABLED=1 /tmp/ebitenmobile-18041009/bin/gobind -lang=go,java -outdir=$WORK -javapkg=net.isoftservices.client octopus/parts/webgames/bin/shooters-go-client/src/android github.com/hajimehoshi/ebiten/v2/mobile/ebitenmobileview
> write /tmp/gomobile-work-1926535437/src/go.mod
> mkdir -p $WORK/src
> PWD=$WORK/src GOOS=android GOARCH=arm CC=$ANDROID_HOME/ndk/24.0.8215888/toolchains/llvm/prebuilt/linux-x86_64/bin/armv7a-linux-androideabi16-clang CXX=$ANDROID_HOME/ndk/24.0.8215888/toolchains/llvm/prebuilt/linux-x86_64/bin/armv7a-linux-androideabi16-clang++ CGO_ENABLED=1 GOARM=7 GOPATH=$WORK:$GOPATH go mod tidy -v
> go: no module declaration in go.mod. To specify the module path:
> 	go mod edit -module=example.com/mod
> rm -r -f "$WORK"
> gomobile: go mod tidy -v failed: exit status 1

For some reason a go.mod in a $WORK dir doesn't contain a valid reference to the module.

**Comment by hajimehoshi** [comment]
Does the specified module `octopus/parts/webgames/bin/shooters-go-client/src/android/` have a `go.mod`?

**Reply by kordax** [reply]
Hi, no. 
I cannot create a submodule because this directory depends on a root project's code.

**Reply by hajimehoshi** [reply]
I'm not sure your project structure, but the module name starting with a non-domain part seems suspicious

**Reply by kordax** [reply]
We're using vendor mode with vendor directory.
Root octopus directory contains go.sum and go.mod. 
octopus/parts/webgames/bin/shooters-go-client directory doesn't contain go.mod and is not a submodule.
it's just a directory containing wasm + android client for a root project.

**Reply by hajimehoshi** [reply]
I see. vendoring doesn't work with gomobile/ebitenmobile unfortunately: https://github.com/golang/go/issues/34181
